rust/src/live.rs

```diff
@@ -184,19 +184,22 @@ fn apply_diff(
 fn update_etc(
     repo: &ostree::Repo,
     diff: &crate::ostree_diff::FileTreeDiff,
+    config_diff: &crate::dirdiff::Diff,
     sepolicy: &ostree::SePolicy,
     commit: &str,
     destdir: &openat::Dir,
 ) -> Result<()> {
     let expected_subpath = "/usr";
     // We stripped both /usr and /etc, we need to readd them both
     // for the checkout.
-    fn filtermap_paths(s: &String) -> Option<(Option<PathBuf>, &Path)> {
-        s.strip_prefix("/etc").map(|p| {
-            let p = Path::new(p).strip_prefix("/").expect("prefix");
-            (Some(Path::new("/usr/etc").join(p)), p)
-        })
-    }
+    let filtermap_paths = |s: &String| -> Option<(Option<PathBuf>, PathBuf)> {
```

Comment:
I also had to return an owned `PathBuf` because I couldn't figure out how to make the compiler happy with lifetimes - I think the problem is we might in theory be borrowing content from the captured `config_diff` variable or something?

Reply:
That's my understanding as well. You'd have to pass the `config_diff` as an argument as well, at which point the result will need to live long enough to survive the lifetimes of both inputs. But that would make plugging it into `.filter_map()` more cumbersome, I think. Unless I'm missing something, this version looks fine enough.